File: 1154.一年中的第几天.go
/*
 * @lc app=leetcode.cn id=1154 lang=golang
 *
 * [1154] 一年中的第几天
 */

// @lc code=start
func dayOfYear(date string) (res int) {
	splitString := strings.Split(date, "-")
	year, _ := strconv.Atoi(splitString[0])
	month, _ := strconv.Atoi(splitString[1])
	day, _ := strconv.Atoi(splitString[2])

	daysEachMonth := [12]int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}

	if year%400 == 0 || (year%4 == 0 && year%100 != 0) {
		daysEachMonth[1]++
	}
	for i := 0; i < month-1; i++ {
		res += daysEachMonth[i]
	}
	res += day
	return
}

// @lc code=end

File: 1154.一年中的第几天.py
#
# @lc app=leetcode.cn id=1154 lang=python3
#
# [1154] 一年中的第几天
#

# @lc code=start
class Solution:
    def dayOfYear(self, date: str) -> int:
        # 拆出年月日
        year, month, day = [int(x) for x in date.split("-")]

        # 模拟每一年的每月天数
        amount = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
        # 闰年二月加一天
        if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
            amount[1] += 1
        # 加总前几个月并加上本月的天数
        return sum(amount[:month - 1]) + day
# @lc code=end

